streaming/java/streaming-runtime/src/main/java/io/ray/streaming/runtime/barrier/Barrier.java

```diff
@@ -0,0 +1,42 @@
+package io.ray.streaming.runtime.barrier;
+
+import com.google.common.base.MoreObjects;
+import java.io.Serializable;
+import java.util.Objects;
+
+public final class Barrier implements Serializable {
+
+  private final long id;
+
+  public Barrier(long id) {
+    this.id = id;
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    Barrier barrier = (Barrier) o;
+    return id == barrier.id;
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(id);
+  }
+
+  @Override
+  public String toString() {
+    return MoreObjects.toStringHelper(this)
+        .add("id", id)
+        .toString();
+  }
+
+  public long getId() {
```

Comment:
Need more doc for such an important data structure

Reply:
This data structure is useless, I removed it.